### PR TITLE
4️⃣ [ORG-61] Introduce OrganizationDomain and return domains from useOrganization

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

This PR:
- Introduces a new resource called OrganizationDomain
- exposes a list of domains of the above type from the useOrganization hook

### CODE REVIEW

File: packages/types/src/organizationDomain.ts [hunk 1]
```diff
@@ -20,8 +20,21 @@ export interface OrganizationDomainResource extends ClerkResource {
...
   delete: () => Promise<void>;
   update: (params: UpdateOrganizationDomainParams) => Promise<OrganizationDomainResource>;
 }
 
+export type PrepareAffiliationVerificationParams = {
+  affiliationEmailAddress: string;
+};
+
+export type AttemptAffiliationVerificationParams = {
+  emailAddress: string;
```
Comment (chanioxaris) [inline packages/types/src/organizationDomain.ts:37]: 🔧 This should be `code`

File: packages/clerk-js/src/core/resources/OrganizationDomain.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    });
+  };
+
+  delete = (): Promise<void> => {
+    return this._baseDelete({
+      path: `/organizations/${this.organizationId}/domains/${this.id}`,
+    });
+  };
+
+  // static async get({
```
Comment (chanioxaris) [inline packages/clerk-js/src/core/resources/OrganizationDomain.ts]: ❓ Do we need this or we can drop it?

**Review by chanioxaris — APPROVED** [review]

File: packages/clerk-js/src/core/resources/OrganizationDomain.ts (hunk 2 above)
Comment (panteliselef) [reply packages/clerk-js/src/core/resources/OrganizationDomain.ts]: Already dropped, maybe it was visible in the per commit review

File: packages/types/src/organizationDomain.ts (hunk 1 above)
Comment (panteliselef) [reply packages/types/src/organizationDomain.ts]: Correct 🙏

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**